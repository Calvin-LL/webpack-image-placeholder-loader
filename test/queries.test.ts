import WIPLWebpackTestCompiler from "./helpers/WIPLWebpackTestCompiler";

describe.each([4, 5] as const)("v%d queries", (webpackVersion) => {
  it('should be overridden by json query `{"r":255,"g":255,"b":255}`', async () => {
    const compiler = new WIPLWebpackTestCompiler({ webpackVersion });
    const bundle = await compiler.compile({
      loaderOptions: {
        color: { r: 0, g: 0, b: 0 },
        esModule: false,
      },
      fileContentOverride: `__export__ = require('./Macaca_nigra_self-portrait_large.jpg?{"color":{"r":255,"g":255,"b":255}}')`,
    });

    expect(bundle.execute("main.js")).toMatchSnapshot("result");
  });

  it("should be overridden by query size=10", async () => {
    const compiler = new WIPLWebpackTestCompiler({ webpackVersion });
    const bundle = await compiler.compile({
      loaderOptions: {
        size: 1,
        esModule: false,
      },
      fileContentOverride:
        '__export__ = require("./Macaca_nigra_self-portrait_large.jpg?size=10")',
    });

    expect(bundle.execute("main.js")).toMatchSnapshot("result");
  });
});
