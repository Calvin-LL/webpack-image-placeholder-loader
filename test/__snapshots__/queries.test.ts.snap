// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`v4 queries should be overridden by json query \`{"r":255,"g":255,"b":255}\`: result 1`] = `"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAIAAACQd1PeAAAACXBIWXMAAAPoAAAD6AG1e1JrAAAADElEQVQI12P4//8/AAX+Av7czFnnAAAAAElFTkSuQmCC"`;

exports[`v4 queries should be overridden by query size=10: result 1`] = `"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAoAAAAKCAIAAAACUFjqAAAACXBIWXMAAAPoAAAD6AG1e1JrAAAAE0lEQVQY02PILvXAgxhGpbFJAwAU3HOheQ0aNgAAAABJRU5ErkJggg=="`;

exports[`v5 queries should be overridden by json query \`{"r":255,"g":255,"b":255}\`: result 1`] = `"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAIAAACQd1PeAAAACXBIWXMAAAPoAAAD6AG1e1JrAAAADElEQVQI12P4//8/AAX+Av7czFnnAAAAAElFTkSuQmCC"`;

exports[`v5 queries should be overridden by query size=10: result 1`] = `"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAoAAAAKCAIAAAACUFjqAAAACXBIWXMAAAPoAAAD6AG1e1JrAAAAE0lEQVQY02PILvXAgxhGpbFJAwAU3HOheQ0aNgAAAABJRU5ErkJggg=="`;
